contracts/FixedProductMarketMaker.sol

```diff
@@ -276,12 +298,31 @@ contract FixedProductMarketMaker is ERC20, ERC1155TokenReceiver {
 
         conditionalTokens.safeTransferFrom(msg.sender, address(this), positionIds[outcomeIndex], outcomeTokensToSell, "");
 
-        uint feeAmount = returnAmount.mul(fee) / (ONE.sub(fee));
-        mergePositionsThroughAllConditions(returnAmount);
+        uint protocolFeeAmount;
+        uint lpFeeAmount;
 
-        require(collateralToken.transfer(msg.sender, returnAmount), "return transfer failed");
+        uint totalFeeAmount = returnAmount.mul(fee) / (ONE.sub(fee));
+        uint totalReturnAmount = returnAmount;
 
-        emit FPMMSell(msg.sender, returnAmount, feeAmount, outcomeIndex, outcomeTokensToSell);
+        FPMMDeterministicFactory factory = FPMMDeterministicFactory(factoryAddress);
+
+        bool protocolFeeOn = factory.protocolFeeOn();
+        uint8 protocolFeeDenominator = factory.protocolFeeDenominator();
+
+        if (protocolFeeOn && protocolFeeDenominator > 0) {
+            protocolFeeAmount = totalFeeAmount / protocolFeeDenominator;
```

Comment:
We can probably define `protocolFeeAmount` here instead of on line 301 as the variable is not used outside the scope of this if statement